docs/reference/deprecated.md

```diff
@@ -1,5 +1,9 @@
 Gutenberg's deprecation policy is intended to support backwards-compatibility for two minor releases, when possible. The current deprecations are listed below and are grouped by _the version at which they will be removed completely_. If your plugin depends on these behaviors, you must update to the recommended alternative before the noted version.
 
+## 4.0.0
```

Comment:
Should we start including the deprecation note in the relevant package's `CHANGELOG.md` as well? I have been and am inclined to think we should.

Reply:
While I agree it would be great, I'm afraid we're adding more on more stuff to think about for the reviewer/merger.

Reply:
But you agreed to it yesterday 😛 

https://wordpress.slack.com/archives/C5UNMSU4R/p1535567253000100

Honestly, if we wanted to limit where we note, I'd say CHANGELOG would be the more sensible default. A document like this one could theoretically be generated by scanning each CHANGELOG for its deprecation sections.

Reply:
I "accidentally" agreed to it, I never thought about deprecations :P

The reason it's not easy to automate is the version where it will be removed (one mentions the Gutenberg version, the other the npm package version).

Reply:
>(one mentions the Gutenberg version, the other the npm package version).

It's good to raise this actually, since we include the Gutenberg version in the `deprecated` function call despite the Gutenberg plugin not having any meaning of its own in the context of the package alone.

Reply:
There is a very narrow group of people who introduce deprecations so we might have more strict rules which will benefit those who consume those packages. I'm fine with adding a section in the changelog like this:

```md
### Deprecations

- `wp.editor.RichTextProvider` flaged for deprecation. Please use `wp.data.select( 'core/editor' )` methods instead.
```

In this case, it will be a patch release so we probably should include it anyway.

Reply:
>In this case, it will be a patch release so we probably should include it anyway.

For a deprecation, we should make it a minor release:

>Minor version [...] MUST be incremented if any public API functionality is marked as deprecated.

https://semver.org/#spec-item-7

Reply:
Addressed with https://github.com/WordPress/gutenberg/pull/9465/commits/0613c4bf56d310c7fe6415631c358cd3a72d02d0, there are breaking changes so it's going to be major anyway :)

By the way, I really like how useful this changelog becomes.